onnx/test/reference_evaluator_test.py

```diff
@@ -81,6 +81,19 @@
     getenv("ORT_MAX_ONNX_OPSET_SUPPORTED_VERSION", "18")
 )
 
+def skip_if_no_re2(fn):
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        try:
+            import re2
+
+            del re2
```

Comment:
We can use the importlib for this check. Seems to be more recommended as it does not actually import the pakage.

Reply:
of course we can use 
spec = importlib.util.find_spec(package_name)
but then not only for re2... 

We currently use this type of verification for other packages such as pytorch and torchvision. I had based my addition on this

Reply:
I think we can start with re2 since it is new code. Old code can be changed following this. I think breaking the consistency here is ok because it is a better way of doing this check

Reply:
done